>From d6a355859bdf0cbe3820a424bcba9b2ecc0c39a9 Mon Sep 17 00:00:00 2001
From: Thomas Bechtold <thomasbechtold@jpberlin.de>
Date: Sun, 22 Dec 2013 11:40:49 +0100
Subject: [PATCH] apt-pkg/contrib/gpgv.cc: use /tmp as fallback dir

if the directory given by $TMPDIR is not available, use /tmp as fallback.
---
 apt-pkg/contrib/gpgv.cc | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/apt-pkg/contrib/gpgv.cc b/apt-pkg/contrib/gpgv.cc
index f47e7ea..8f619fe 100644
--- a/apt-pkg/contrib/gpgv.cc
+++ b/apt-pkg/contrib/gpgv.cc
@@ -10,6 +10,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <unistd.h>
 
 #include<apt-pkg/configuration.h>
 #include<apt-pkg/error.h>
@@ -22,11 +23,15 @@
 static char * GenerateTemporaryFileTemplate(const char *basename)	/*{{{*/
 {
    const char *tmpdir = getenv("TMPDIR");
+
 #ifdef P_tmpdir
    if (!tmpdir)
       tmpdir = P_tmpdir;
 #endif
-   if (!tmpdir)
+
+   // check that tmpdir is set and exists
+   struct stat st;
+   if (!tmpdir || stat(tmpdir, &st) != 0)
       tmpdir = "/tmp";
 
    std::string out;
-- 
1.8.5.2

